protocol/x/clob/keeper/liquidations.go

```diff
@@ -99,9 +111,13 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
 			if numAttemptedDeleveraging < k.MaxDeleveragingAttemptsPerBlock {
 				// The liquidation order was unfilled. Try to deleverage the subaccount.
 				subaccountId := liquidationOrder.GetSubaccountId()
-				perpetualId := liquidationOrder.MustGetLiquidatedPerpetualId()
 				deltaQuantums := liquidationOrder.GetDeltaQuantums()
 
+				// Get a pseudorandom perpetualId from the list of open positions.
+				subaccount := k.subaccountsKeeper.GetSubaccount(ctx, subaccountId)
+				positions := subaccount.GetPerpetualPositions()
+				perpetualId := positions[pseudoRand.Intn(len(positions))].PerpetualId
```

Comment:
I think a panic here is justified since [this liquidation order validation](https://github.com/dydxprotocol/v4-chain/blob/a68b735cbe6031fd3dae3dfc9870dbe0ff7ce8e3/protocol/x/clob/keeper/liquidations.go#L152-L155) must have passed for this logic to be executed.

Understand this is an implicit dependency though. Maybe worth adding a comment that this code assumes that the subaccount has at least one open position? Would prefer to reduce the number of errors returned upstream since it makes the code unreadable + errors in the liquidation flow are usually worth halting the chain for.